bundles/org.eclipse.swt.svg/src/org/eclipse/swt/svg/JSVGRasterizer.java

```diff
@@ -0,0 +1,144 @@
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Michael Bangas (Vector Informatik GmbH) - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.swt.svg;
+
+import static java.awt.RenderingHints.KEY_ALPHA_INTERPOLATION;
+import static java.awt.RenderingHints.KEY_ANTIALIASING;
+import static java.awt.RenderingHints.KEY_COLOR_RENDERING;
+import static java.awt.RenderingHints.KEY_DITHERING;
+import static java.awt.RenderingHints.KEY_FRACTIONALMETRICS;
+import static java.awt.RenderingHints.KEY_INTERPOLATION;
+import static java.awt.RenderingHints.KEY_RENDERING;
+import static java.awt.RenderingHints.KEY_STROKE_CONTROL;
+import static java.awt.RenderingHints.KEY_TEXT_ANTIALIASING;
+import static java.awt.RenderingHints.VALUE_ALPHA_INTERPOLATION_QUALITY;
+import static java.awt.RenderingHints.VALUE_ANTIALIAS_ON;
+import static java.awt.RenderingHints.VALUE_COLOR_RENDER_QUALITY;
+import static java.awt.RenderingHints.VALUE_DITHER_DISABLE;
+import static java.awt.RenderingHints.VALUE_FRACTIONALMETRICS_ON;
+import static java.awt.RenderingHints.VALUE_INTERPOLATION_BICUBIC;
+import static java.awt.RenderingHints.VALUE_RENDER_QUALITY;
+import static java.awt.RenderingHints.VALUE_STROKE_PURE;
+import static java.awt.RenderingHints.VALUE_TEXT_ANTIALIAS_ON;
+
+import java.awt.Graphics2D;
+import java.awt.RenderingHints.Key;
+import java.awt.image.BufferedImage;
+import java.awt.image.DataBufferInt;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.graphics.ImageData;
+import org.eclipse.swt.graphics.PaletteData;
+import org.eclipse.swt.internal.image.SVGRasterizer;
+
+import com.github.weisj.jsvg.SVGDocument;
+import com.github.weisj.jsvg.geometry.size.FloatSize;
+import com.github.weisj.jsvg.parser.LoaderContext;
+import com.github.weisj.jsvg.parser.SVGLoader;
+
+/**
+ * A rasterizer implementation for converting SVG data into rasterized images.
+ * This class uses the third party library JSVG for the raterization of SVG
+ * images.
+ * 
+ * @since 1.0.0
+ */
+public class JSVGRasterizer implements SVGRasterizer {
+
+	private static final SVGLoader SVG_LOADER = new SVGLoader();
+
+	private final static Map<Key, Object> RENDERING_HINTS = Map.of( //
+			KEY_ANTIALIASING, VALUE_ANTIALIAS_ON, //
+			KEY_ALPHA_INTERPOLATION, VALUE_ALPHA_INTERPOLATION_QUALITY, //
+			KEY_COLOR_RENDERING, VALUE_COLOR_RENDER_QUALITY, //
+			KEY_DITHERING, VALUE_DITHER_DISABLE, //
+			KEY_FRACTIONALMETRICS, VALUE_FRACTIONALMETRICS_ON, //
+			KEY_INTERPOLATION, VALUE_INTERPOLATION_BICUBIC, //
+			KEY_RENDERING, VALUE_RENDER_QUALITY, //
+			KEY_STROKE_CONTROL, VALUE_STROKE_PURE, //
+			KEY_TEXT_ANTIALIASING, VALUE_TEXT_ANTIALIAS_ON //
+	);
+
+	@Override
+	public ImageData[] rasterizeSVG(InputStream inputStream, int zoom) throws IOException {
```

Comment:
Couldn't this method just return a single ImageData instead of an array? Is there any scenario in which it will return multiple data?

Reply:
done.